dbt/include/databricks/macros/relations/tblproperties.sql

```diff
@@ -3,8 +3,12 @@
 {%- endmacro -%}
 
 {% macro databricks__tblproperties_clause() -%}
-  {%- set tblproperties = config.get('tblproperties') -%}
-  {%- if tblproperties is not none %}
+  {%- set tblproperties = config.get('tblproperties', {}) -%}
+  {%- set is_iceberg = adapter.check_iceberg(config.get('table_format'), config.get('file_format'), model.config.materialized) -%}
+  {%- if is_iceberg -%}
+    {%- set _ = tblproperties.update({'delta.enableIcebergCompatV2': 'true', 'delta.universalFormat.enabledFormats': 'iceberg'}) -%}
```

Comment:
It seems the update logic is duplicate, I'm not sure if it is possible but can it be extracted to a central place maybe in the adapter.updateTblProperties()?

Reply:
Good call, not sure why I decided the iceberg check should be in the adapter but not updating tblproperties.